backends/tfhe-cuda-backend/build.rs

```diff
@@ -51,6 +54,53 @@ fn main() {
         println!("cargo:rustc-link-lib=cudart");
         println!("cargo:rustc-link-search=native=/usr/lib/x86_64-linux-gnu/");
         println!("cargo:rustc-link-lib=stdc++");
+
+        let header_path = "wrapper.h";
+        println!("cargo:rerun-if-changed={}", header_path);
+
+        let out_path = PathBuf::from("src").join("bindings.rs");
+
+        // Check modification times
+        let header_modified = fs::metadata(header_path).unwrap().modified().unwrap();
+        let bindings_modified = if out_path.exists() {
+            fs::metadata(&out_path).unwrap().modified().unwrap()
+        } else {
+            std::time::SystemTime::UNIX_EPOCH // If bindings file doesn't exist, consider it older
+        };
+
+        // Regenerate bindings only if header has been modified
+        if header_modified > bindings_modified {
```

Comment:
I'm not sure I follow, isn't the rerun-if-changed enough ?

Reply:
Let me check

Reply:
Looks like it's enough indeed, makes things simpler!

Reply:
Hmm no wait, if I remove this part I get an empty change detected by git on the file again...

Reply:
what ? 🤔

Reply:
:shrug:

Reply:
an empty change ?

Reply:
I don't get it sorry 😅

Reply:
Oh it's probably the time stamp that gets updated so git sees the file as modified, even though the content of the file has not changed.

Reply:
so weird, I sometimes modify files and then save them unchanged in the end but I don't get a diff for that :| git is weird then I guess

Reply:
Let me try to see if the ci passes when I remove this part. I see an empty change locally but maybe the ci is fine :thinking:

Reply:
Nope, it fails: https://github.com/zama-ai/tfhe-rs/actions/runs/11295491506/job/31418412019?pr=1659

Reply:
I'm putting back this part